module/documents/activity/attack.mjs

```diff
@@ -189,6 +193,11 @@ export default class AttackActivity extends ActivityMixin(AttackActivityData) {
     if ( ammo ) {
       ammoUpdate = { id: ammo.id, quantity: Math.max(0, ammo.system.quantity - 1) };
       ammoUpdate.destroy = ammo.system.uses.autoDestroy && (ammoUpdate.quantity === 0);
+      await this.item.setFlag("dnd5e", `last.${this.id}.ammunition`, roll.options.ammunition);
+    }
+
+    if ( roll.options.attackMode ) {
+      await this.item.setFlag("dnd5e", `last.${this.id}.attackMode`, roll.options.attackMode);
```

Comment:
That's a lot of updates and rendering of sheets to prepopulate a dialog. Doesn't it make more sense to use a local registry object? E.g.,
```js
AttackActivity.#storage.set(this.uuid, {ammo: ..., attack: ...});
```